engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/NameRecordingScreen.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layers.mainMenu;
+
+import org.codehaus.plexus.util.FileUtils;
+import org.codehaus.plexus.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.config.Config;
+import org.terasology.engine.GameEngine;
+import org.terasology.engine.modes.StateLoading;
+import org.terasology.engine.paths.PathManager;
+import org.terasology.game.GameManifest;
+import org.terasology.i18n.TranslationSystem;
+import org.terasology.network.NetworkMode;
+import org.terasology.recording.RecordAndReplayUtils;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.animation.MenuAnimationSystems;
+import org.terasology.rendering.nui.layers.mainMenu.savedGames.GameInfo;
+import org.terasology.rendering.nui.widgets.UIButton;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.rendering.nui.widgets.UIText;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Screen for setting the name of and ultimately loading a recording.
+ */
+public class NameRecordingScreen extends CoreScreenLayer {
+    public static final ResourceUrn ASSET_URI = new ResourceUrn("engine:nameRecordingScreen!instance");
+
+    private static final Logger logger = LoggerFactory.getLogger(NameRecordingScreen.class);
+
+    @In
+    protected Config config;
+
+    @In
+    private TranslationSystem translationSystem;
+
+    private GameInfo gameInfo;
+
+    private RecordAndReplayUtils recordAndReplayUtils;
+
+    // widgets
+    private UILabel title;
```

Comment:
Checkstyle gives this warning: "Private field 'title' is assigned but never accessed"

Reply:
Removed and fixed :)